cluster.go

```diff
@@ -32,6 +33,17 @@ func (cl *Cluster) NextID(ctx context.Context) (int, error) {
 	return strconv.Atoi(ret)
 }
 
+func (cl *Cluster) IsVMIDTaken(ctx context.Context, vmid int) (bool, error) {
```

Comment:
i love this PR overall but not a huge fan of the name of this func, i think i'd like to see it more of a <something>ID so it's like NextID... how do you feel about CheckID or maybe ValidID? open to other ideas

Reply:
I don't like the name either. I think `validID` could be confused with "Is this a valid ID" in terms of format.
The API documentation describes the function as

> Pass a VMID to assert that its free (at time of check).

I renamed the function to `checkID` and inverted the return value logic (so `true` means "free" and `false` means "taken"). I think `checkID` is the best idea so far... if you/someone comes up with a better idea I'm happy to rename it.

Reply:
i can live with this, let's get it merged